evm/src/cpu/kernel/asm/core/terminate.asm

```diff
@@ -95,14 +96,9 @@ global sys_selfdestruct:
     // stack: balance_ptr, 0, balance, address, recipient, kexit_info
     %mstore_trie_data
 
+    // Send the balance to the recipient. 
     %stack (balance, address, recipient, kexit_info) ->
-        (address, recipient, address, recipient, balance, kexit_info)
-
-    // If the recipient is the same as the address, then we're done.
-    // Otherwise, send the balance to the recipient.
-    // stack: address, recipient, address, recipient, balance, kexit_info
-    EQ %jumpi(sys_selfdestruct_journal_add)
```

Comment:
Don't we still need this in case we're in a selfdestruct of a contract that was created in the same txn?
Looking at this in the EIP
<img width="888" alt="image" src="https://github.com/0xPolygonZero/plonky2/assets/22958988/72275bae-68da-4118-ac26-d776a3b99ee2">

Reply:
If we're in that case, the account will be in the selfdestruct list and will be deleted anyway, which erases all of the node data (oncluding its balance).
If we keep these lines, there will be a discrepancy in case of revertion.

Reply:
Right, but what if we call `BALANCE` on this address before the end of the tx?

Reply:
Oops, you're right! I'll have to put this `add_eth` inside the "has_been_created" logic in `maybe_insert_selfdestruct_list`.